build_tools/cmake/iree_e2e_matmul_test.cmake

```diff
@@ -216,12 +216,6 @@ function(iree_single_backend_e2e_matmul_test)
   if(DEFINED _RULE_DRIVER)
     string(TOUPPER ${_RULE_DRIVER} _UPPERCASE_DRIVER)
     string(REPLACE "-" "_" _NORMALIZED_DRIVER ${_UPPERCASE_DRIVER})
-    string(TOUPPER "${IREE_EXTERNAL_HAL_DRIVERS}" _UPPERCASE_EXTERNAL_DRIVERS)
```

Comment:
@ScottTodd as you commented in https://github.com/openxla/iree/pull/16466#discussion_r1494788870, this is an issue for external targets that are not explicitly turned on. So need to remove it.

Reply:
I'm not sure I follow ... if the check below (lines 219-220) works then I don't see why this wouldn't also work?

Fine to remove the overly aggressive `SEND_ERROR` regardless though. Might want to do that in `iree_check_test.cmake` also for consistency.

Reply:
Okay dropped the `SEND_ERROR` in `iree_check_test.cmake` too for consistency.